File: prayer_app_expo/src/components/AppHeader.js
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { Spacing, SalahLayout, interFont } from '../theme/theme';
import { useTheme } from '../providers/ThemeProvider';
import AppIconButton from './AppIconButton';

export default function AppHeader({ title, onSettingsTap, onTestNotification }) {
    const { theme: tc } = useTheme();
    return (
        <View style={styles.container}>
            <View style={styles.locationRow}>
                <MaterialCommunityIcons
                    name="map-marker"
                    size={SalahLayout.locationIconSize}
                    color={tc.accent}
                />
                <Text style={[styles.title, { color: tc.textPrimary }]}>{title}</Text>
            </View>
            <View style={styles.buttonRow}>
                {onTestNotification && (
                    <AppIconButton
                        icon="bell-outline"
                        size={SalahLayout.gearButtonSize}
                        iconSize={SalahLayout.gearIconSize}
                        onPress={onTestNotification}
                    />
                )}
                {onTestNotification && <View style={{ width: 4 }} />}
                <AppIconButton
                    icon="cog-outline"
                    size={SalahLayout.gearButtonSize}
                    iconSize={SalahLayout.gearIconSize}
                    onPress={onSettingsTap || (() => { })}
                />
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
        paddingHorizontal: SalahLayout.screenPadding,
    },
    locationRow: {
        flexDirection: 'row',
        alignItems: 'center',
        gap: Spacing.s8,
    },
    buttonRow: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    title: {
        fontFamily: interFont('500'),
        fontSize: 16,
    },
});

File: prayer_app_flutter/lib/src/components/app_header.dart
import 'package:flutter/material.dart';
import 'package:material_design_icons_flutter/material_design_icons_flutter.dart';
import '../theme/app_theme.dart';
import '../providers/theme_provider.dart';
import 'app_icon_button.dart';

class AppHeader extends StatelessWidget {
  final String title;
  final VoidCallback? onSettingsTap;
  final VoidCallback? onTestNotification;

  const AppHeader({
    super.key,
    required this.title,
    this.onSettingsTap,
    this.onTestNotification,
  });

  @override
  Widget build(BuildContext context) {
    final tc = ThemeScope.of(context).current;
    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: SalahLayout.screenPadding),
      child: Row(
        mainAxisAlignment: MainAxisAlignment.spaceBetween,
        children: [
          Row(
            children: [
              Icon(
                MdiIcons.mapMarker,
                color: tc.accent,
                size: SalahLayout.locationIconSize,
              ),
              const SizedBox(width: AppSpacing.s8),
              Text(
                title,
                style: AppTypography.body(tc).copyWith(
                  fontSize: 16,
                  fontWeight: FontWeight.w500,
                ),
              ),
            ],
          ),
          Row(
            mainAxisSize: MainAxisSize.min,
            children: [
              if (onTestNotification != null)
                AppIconButton(
                  icon: MdiIcons.bellOutline,
                  size: SalahLayout.gearButtonSize,
                  iconSize: SalahLayout.gearIconSize,
                  onTap: onTestNotification!,
                ),
              if (onTestNotification != null)
                const SizedBox(width: 4),
              AppIconButton(
                icon: MdiIcons.cogOutline,
                size: SalahLayout.gearButtonSize,
                iconSize: SalahLayout.gearIconSize,
                onTap: onSettingsTap ?? () {},
              ),
            ],
          ),
        ],
      ),
    );
  }
}

